lib/stats/service.js

```diff
@@ -0,0 +1,56 @@
+const {groupBy} = require('lodash')
+const {format, compareDesc, add, endOfDay} = require('date-fns')
+
+function getCumulFirstRevisionsByDate(firstRevisions, dates) {
+  const cumulFirstRevisionsByDate = []
+  for (
+    let dateIterator = endOfDay(new Date(dates.dateFrom.getTime()));
+    compareDesc(dateIterator, dates.dateTo) > 0;
+    dateIterator = add(dateIterator, {days: 1})
+  ) {
+    cumulFirstRevisionsByDate.push({
+      date: format(dateIterator, 'yyyy-MM-dd'),
+      totalCreations: firstRevisions.filter(firstRevision => compareDesc(firstRevision.publishedAt, dateIterator) === 1).length
+    })
+  }
+
+  return cumulFirstRevisionsByDate
+}
+
+function getBalsByDays(revisions) {
+  const revisionsGroupByDays = groupBy(revisions, revision =>
+    format(revision.publishedAt, 'yyyy-MM-dd')
+  )
+  const balsByDays = []
+  for (const date in revisionsGroupByDays) {
+    if (Object.hasOwnProperty.call(revisionsGroupByDays, date)) {
+      const revisions = revisionsGroupByDays[date]
+
+      const revisionsGroupByBals = groupBy(revisions, revision =>
+        revision.codeCommune
+      )
+      const publishedBAL = []
+      for (const codeCommune in revisionsGroupByBals) {
+        if (Object.hasOwnProperty.call(revisionsGroupByBals, codeCommune)) {
+          const revisionsByBal = revisionsGroupByBals[codeCommune]
+          publishedBAL.push({
+            codeCommune,
+            numPublications: revisionsByBal.length
+          })
+        }
+      }
+
+      balsByDays.push({
+        date,
+        publishedBAL
+      })
+    }
+  }
+
+  return balsByDays
+}
```

Comment:
J'avoue que je trouve les `for…in` toujours un peu illisible, surtout avec les vérifications assez lourdes demandées par le linter comme `Object.hasOwnProperty.call`.

Qu'est-ce que tu en penses ?

```suggestion
function getBalsByDays(revisions) {
  const revisionsGroupByDays = groupBy(revisions, revision =>
    format(revision.publishedAt, 'yyyy-MM-dd')
  )

  const balsByDays = Object.entries(revisionsGroupByDays).map(([date, revisions]) => {
    const revisionsGroupByBals = groupBy(revisions, revision =>
      revision.codeCommune
    )

    const publishedBAL = Object.entries(revisionsGroupByBals).map(([codeCommune, revisionsByBal]) => ({
      codeCommune,
      numPublications: revisionsByBal.length
    }))

    return {
      date,
      publishedBAL
    }
  })

  return balsByDays
}
```